channeldb/kvdb/etcd/db.go

```diff
@@ -130,18 +131,40 @@ type BackendConfig struct {
 	// Pass is the password for the etcd peer.
 	Pass string
 
+	// CertFile holds the path to the TLS certificate for etcd RPC.
+	CertFile string
+
+	// KeyFile holds the path to the TLS private key for etcd RPC.
+	KeyFile string
```

Comment:
Odd that we would need the _private_ key here right? In my local branch, seemed it wouldn't proceed at runtime unless this was set. Perhaps we need to specify some custom validation for the cert. Note that this is only needed for self signed certs, if etcd host as a domain, and it has a "real" cert, then it'll validate as normal.

Reply:
PTAL

Reply:
In the end I reverted back to the way `etcd` does the TLS config as that actually seems to be correct to me. Could do TLS with auto generated certs and insecure skip verify. I'm not sure why the trusted cert requires the private key or if it requires at all, but please also try with `etcdctl --debug` to see if that works and what error it gives.